homeassistant/components/stt/legacy.py

```diff
@@ -0,0 +1,169 @@
+"""Handle legacy speech to text platforms."""
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from collections.abc import AsyncIterable, Coroutine
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import config_per_platform, discovery
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.setup import async_prepare_setup_platform
+
+from .const import (
+    DOMAIN,
+    AudioBitRates,
+    AudioChannels,
+    AudioCodecs,
+    AudioFormats,
+    AudioSampleRates,
+    SpeechResultState,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def async_get_provider(
```

Comment:
I'm thinking we'll have a different helper to retrieve the entity from the component. So I consider `async_get_provider` legacy.